scipy/optimize/tests/test_optimize.py

```diff
@@ -157,7 +157,9 @@ def test_bfgs_nan(self):
         fprime = lambda x: np.ones_like(x)
         x0 = [np.nan]
         with np.errstate(over='ignore', invalid='ignore'):
-            x = optimize.fmin_bfgs(func, x0, fprime, disp=False)
+            with warnings.catch_warnings():
+                warnings.simplefilter('ignore', LineSearchWarning)
+                x = optimize.fmin_bfgs(func, x0, fprime, disp=False)
```

Comment:
Nikolay Mayorov wrote:
> But if we filter warnings in |_line_search_wolfe12| a user won't see
> them when this wrapper is called. It looks like undesired behaviour,
> doesn't it?

No, because this wrapper handles the case of line_search_wolfe2 
returning None and then raise an exception (which is then catched in the 
caller). So there's no point warning the user about something they have 
no control on.

> I might be confused with |LineSearchWarning| vs |_LineSearchError|. The
> former is raised in |line_search_wolfe2| when |derphi_star| is |None|,
> the latter is raised in |_line_search_wolfe12| when |alpha_star| is |None|.
>
> As far as I understand we still want to give a warning (but not an
> error), when |derphi_star| is |None|. Although currently I have no idea
> about algorithm details.
>

I think we want a warning only when line_search_wolfe2 is called 
directly by the user, to let them know about converge issue, although 
the function returns something (this behavior must be kept for backwards 
compatibility).